interfaces/inc/c_pal/uuid.h

```diff
@@ -4,31 +4,54 @@
 #ifndef UUID_H
 #define UUID_H
 
+#ifdef __cplusplus
+#include <cinttypes>
+#include <cstdbool>
+#include <cstdalign>
+#else
 #include <inttypes.h>
 #include <stdbool.h>
+#include <stdalign.h>
+#endif
 
 #ifdef WIN32
 #include "windows.h"
 #endif
 
-#define UUID_T_LENGTH       16
-typedef unsigned char UUID_T[UUID_T_LENGTH]; /*introduces UUID_T as "array of 16 bytes"*/
+#include "macro_utils/macro_utils.h"
 
-typedef const UUID_T const_UUID_T;
+typedef struct UUID_T_TAG
+{
+    alignas(8) uint8_t bytes[16];

```

Comment:
>alignas(8) [](http://example.com/codeflow?start=4&length=10)

Does this create now potentially interesting gaps in some of our structures which would mean more memory usage?

Reply:
Potentially yes.
It would need to be a structure like...

struct TEST
{
char t;
UUID_T etag;
}

The gap here would be of 7 bytes, which is the maximum possible.

Reply:
Also memory *usage* is interesting, because 

struct A
{ char a;}

and struct B
{uin64_t b;}

actually eat the same amount of dynamically allocated memory.

So "interesting". I suppose the size of the structure should be such as the +bytes induced by the alignment would have to cross some minimal allocation a boundary.

Reply:
This can always be mitigated by re-arranging the fields in teh struct.